source/extensions/filters/http/stateful_session/stateful_session.h

```diff
@@ -66,11 +87,50 @@ class StatefulSession : public Http::PassThroughFilter,
   // Http::StreamEncoderFilter
   Http::FilterHeadersStatus encodeHeaders(Http::ResponseHeaderMap& headers, bool) override;
 
+  // Http::StreamFilterBase
+  Http::LocalErrorStatus onLocalReply(const Http::StreamFilterBase::LocalReplyData&) override {
+    // For strict mode, if an override was attempted and a local reply was sent (e.g., 503),
+    // consider this a failed-closed selection.
+    if (override_attempted_ && effective_config_ != nullptr && effective_config_->isStrict() &&
+        !accounted_) {
+      markFailedClosed();
+      accounted_ = true;
+    }
+    return Http::LocalErrorStatus::Continue;
+  }
+
   Http::SessionStatePtr& sessionStateForTest() { return session_state_; }
 
 private:
+  void markOverrideAttempted() { override_attempted_ = true; }
+  void markRouted() {
+    if (override_attempted_ && effective_config_ != nullptr) {
+      if (auto stats = effective_config_->stats(); stats.has_value()) {
+        stats->routed_.inc();
+      }
+    }
+  }
+  void markFailedClosed() {
+    if (override_attempted_ && effective_config_ != nullptr) {
+      if (auto stats = effective_config_->stats(); stats.has_value()) {
+        stats->failed_closed_.inc();
+      }
+    }
+  }
+  void markFailedOpen() {
+    if (override_attempted_ && effective_config_ != nullptr) {
+      if (auto stats = effective_config_->stats(); stats.has_value()) {
+        stats->failed_open_.inc();
+      }
+    }
+  }
```

Comment:
is `if (override_attempted_ && effective_config_ != nullptr)` necessary? I think when we calling these method, all these conditions should always be true.

Reply:
Yep, not needed. I'll remove it. Thanks for calling it out.